### PR TITLE
A new rule for automatic reconciliation and a cron [rebased]

### PR DESCRIPTION
**Pull request by lepistone**
A rebased version of https://github.com/OCA/bank-statement-reconcile/pull/7

Discussion continues from there.

### CODE REVIEW

File: account_advanced_reconcile_bank_statement/advanced_reconciliation.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            'name',
+            'partner_id',
+            'account_id',
+            'move_id',
+            'statement_id')
+        result = ["account_move_line.%s" % col for col in aml_cols]
+        result += ["account_bank_statement.name as statement_name"]
+        return result
+
+    def _from(self, rec, *args, **kwargs):
```
Comment (guewen) [inline account_advanced_reconcile_bank_statement/advanced_reconciliation.py:40]: It should extend and not override (I'll make the change).

File: account_advanced_reconcile_bank_statement/advanced_reconciliation.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+               (move_line['statement_name'] or '').lower().strip())
+
+    # Re-defined for this particular rule; since the commission line is a
+    # credit line inside of the target statement, it should also be considered
+    # as an opposite to be reconciled.
+    def _action_rec(self, cr, uid, rec, context=None):
+        credit_lines = self._query_credit(cr, uid, rec, context=context)
+        debit_lines = self._query_debit(cr, uid, rec, context=context)
+        return self._rec_auto_lines_advanced(
+                cr, uid, rec, credit_lines, credit_lines + debit_lines,
```
Comment (guewen) [inline account_advanced_reconcile_bank_statement/advanced_reconciliation.py]: Shouldn't it be

```
credit_lines + debit_lines,
credit_lines + debit_lines,
```

according to https://github.com/guewen/bank-statement-reconcile/commit/279a176e22bcca4db214dae665abafb02cfd672d

**Comment by nbessi** [comment]
:+1:

**Comment by bwrsandman** [comment]
Needs to be rebased

**Comment by guewen** [comment]
Rebased

**Comment by lepistone** [comment]
rebased again

**Comment by lepistone** [comment]
A committed a few trivial fixes. The original author was @mdietrichc2c, and I vote :+1: !

@guewen you fixed something here, too. Do you approve as it stands now?

Thanks!

**Comment by guewen** [comment]
Aye :+1:

**Comment by lepistone** [comment]
Thanks. We have 3 approvals other than the original author, so I merge.

**Pull request merged**